client/my-sites/marketing/connections/connection.jsx

```diff
@@ -239,6 +239,10 @@ class SharingConnection extends Component {
 
 const SharingConnectionKeyringUserLabel = localize(
 	( { siteId, keyringUserId, translate, userId } ) => {
+		if ( ! keyringUserId ) {
```

Comment:
Without this check, we'll end up with all users from the site and will pick the first one from the list, which is completely incorrect.